File: Algorithms/3741. Minimum Distance Between Three Equal Elements II.ts
function minimumDistance(nums: number[]): number {
    let freq: Map<number, number[]> = new Map()
    for (let i = 0; i < nums.length; i++) {
        let val = nums[i]
        if (!freq.has(val)) {
            freq.set(val, [])
        }
        freq.get(val)!.push(i)
    }
    let ans: number = Number.MAX_SAFE_INTEGER

    for (const [key, value] of freq) {
        if (value.length <= 2) continue
        for (let x = 2; x < value.length; x++) {
            let i = value[x-2]
            let j = value[x-1]
            let k = value[x]
            let calc = Math.abs(i - j) + Math.abs(i - k) + Math.abs(j - k)
            console.log(i, j, k, calc)
            ans = Math.min(ans, calc)
        }
    }
    return ans === Number.MAX_SAFE_INTEGER ? -1 : ans
};

File: Algorithms/3741. Minimum Distance Between Three Equal Elements II.cpp
class Solution {
public:
    int minimumDistance(vector<int>& nums) {
        map<int, vector<int>> mp; 
        for (int i = 0; i < nums.size(); i++) {
            mp[nums[i]].push_back(i);
        }

        int ans = INT_MAX;
        for (auto [k, v] : mp) {
            if (v.size() < 3) continue;
            // for (auto val : v) cout << k << " " << val << endl;
            for (int x = 2; x < v.size(); x++) {
                int i = v[x-2];
                int j = v[x-1];
                int k = v[x];
                int calc = abs(i-j) + abs(k-j) + abs(k-i);
                ans = min(ans, calc);
            }
        }
        return ans == INT_MAX ? -1 : ans;
    }
};